### PR TITLE
DSG-2781 fix permissions issues in invitations

### PR DESCRIPTION
**Pull request by afcalderonm**
## Description
change the logic to authorize public user to delete or post invitations

## Motivation and Context
it is required because it was blocking public users linked to profiles as admin to create invites or delete them

## How Has This Been Tested?
tested writing unit test and locally using the employer portal

## Screenshots (if appropriate):

invite a new user as a public user linked to a profile as admin
image

image

resending invitation
image
image

deleting invitation
image
image

## Types of changes

Patch
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation change (documentation only, will not require redeployment)
- [ ] Refactor or cleanup (functionality unchanged)
Minor
- [ ] New feature (non-breaking change which adds functionality)
Major
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

- [x] I have confirmed all acceptance criteria defined by the story have been met

### CODE REVIEW

File: service/src/main/java/io/nuvalence/workmanager/service/controllers/EmployerProfilesApiDelegateImpl.java [hunk 1]
```diff
@@ -590,4 +578,23 @@ private void postAuditEventForProfileInvitation(
...
+            Optional<EmployerUserLink> profilesLinkOptional =
+                    employerUserLinkService.getEmployerUserLink(profileId, UUID.fromString(userId));
+
+            if (profilesLinkOptional.isEmpty()
+                    || !profilesLinkOptional
+                            .get()
+                            .getProfileAccessLevel()
+                            .equals(ProfileAccessLevel.ADMIN)) {
+                throw new ForbiddenException(
+                        "User is not authorized to invite users to this profile");
```
Comment (jgnuvalence) [inline service/src/main/java/io/nuvalence/workmanager/service/controllers/EmployerProfilesApiDelegateImpl.java:596]: in the case of deleting this message is not accurate

File: service/src/main/java/io/nuvalence/workmanager/service/controllers/IndividualProfilesApiDelegateImpl.java [hunk 2]
```diff
@@ -608,4 +594,24 @@ private void postAuditEventForProfileInvitation(
...
+                    individualUserLinkService.getIndividualUserLinkByProfileAndUserId(
+                            profileId, UUID.fromString(userId));
+
+            if (profilesLinkOptional.isEmpty()
+                    || !profilesLinkOptional
+                            .get()
+                            .getAccessLevel()
+                            .equals(ProfileAccessLevel.ADMIN)) {
+                throw new ForbiddenException(
+                        "User is not authorized to invite users to this profile");
```
Comment (jgnuvalence) [inline service/src/main/java/io/nuvalence/workmanager/service/controllers/IndividualProfilesApiDelegateImpl.java:613]: in the case of deleting this message is not accurate

**Review by jgnuvalence — CHANGES_REQUESTED** [review]

File: service/src/main/java/io/nuvalence/workmanager/service/controllers/IndividualProfilesApiDelegateImpl.java (hunk 2 above)
Comment (afcalderonm) [reply service/src/main/java/io/nuvalence/workmanager/service/controllers/IndividualProfilesApiDelegateImpl.java]: thanks julian, i just modify the code to implement specific exepction message

File: service/src/main/java/io/nuvalence/workmanager/service/controllers/EmployerProfilesApiDelegateImpl.java (hunk 1 above)
Comment (afcalderonm) [reply service/src/main/java/io/nuvalence/workmanager/service/controllers/EmployerProfilesApiDelegateImpl.java]: thanks julian, i just modify the code to implement specific exepction message

**Review by jgnuvalence — APPROVED** [review]

**Pull request merged**